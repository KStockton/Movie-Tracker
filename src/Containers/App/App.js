import React from "react";
import Main from "../Main/Main";
import { Switch, Route, Redirect } from "react-router-dom";
import UserSignIn from "../UserSignIn/UserSignIn";
import { connect } from "react-redux";
import Nav from "../Nav/Nav";
import { Movie } from "../../Components/Movie/Movie";
import Favorites from "../Favorites/Favorites.js";


export class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Nav />
        <Switch>
          <Route exact path="/" component={Main} />
          <Route path="/login" component={UserSignIn} />
          <Route path="/movie/:id" component={Movie} />
          <Route path="/favorites" component={Favorites} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(App);
