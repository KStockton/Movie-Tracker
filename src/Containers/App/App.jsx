import React from "react";
import Main from "../Main/Main.jsx";
import { Switch, Route } from "react-router-dom";
import UserSignIn from "../UserSignIn/UserSignIn";
import { connect } from "react-redux";
import Nav from '../Nav/Nav'

class App extends React.Component {
  render() {
    console.log("rendering");
    return (
      <div className="App">
      <Nav />
        <Switch>
          <Route exact path="/" component={Main} />
          <Route path="/login" component={UserSignIn} />
          <Route path="/user-settings" component={UserSignIn} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(App);
