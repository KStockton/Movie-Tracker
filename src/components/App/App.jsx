import React from "react";
import Main from "../Main/Main.jsx";
import { Switch, Route, Redirect } from "react-router-dom";
import Home from "../Home/Home";
import UserSignIn from "../UserSignIn/UserSignIn";
import UserSettings from "../UserSettings/UserSettings";
import { connect } from "react-redux";

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route
            exact
            path="/"
            render={() =>
              this.props.user.name === undefined ? (
                <Redirect to="/login"/>
              ) : (
                <Main />
              )
            }
          />
          <Route exact path="/" component={Main} />
          <Route path="/login" component={UserSignIn} />
          <Route path="/user-settings" component={UserSettings} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(App);
