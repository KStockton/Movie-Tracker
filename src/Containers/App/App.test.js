import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <Provider store={store}>
        <App/>
      </Provider>
    )
  })

  it('App component should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
