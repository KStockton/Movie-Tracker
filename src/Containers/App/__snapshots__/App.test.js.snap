// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `undefined`;
